### ISSUE TITLE
Reply from person I'm not following is appearing in Following feed

### ISSUE THREAD

**Issue by tobiasvl** [bluesky-social/social-app#7049]
### Steps to Reproduce

A reply chain `A -> B -> C` appears in my Following feed. I follow A and C, but I do not follow B.

![image](https://github.com/user-attachments/assets/33c67b39-22db-49ca-a76f-0d611e994cd4)

It's my understanding that #5049 introduced `A -> B -> A` to the Following feed (if a "number of likes" criterion was met), and that #5301 reverted this again.

It seems counter-intuitive to me that `A -> B -> A` should not appear, but that `A -> B -> C` (where `A != C`) should?

This seems like a dupe/special case of #1762, but that's closed.

### What platform(s) does this occur on?

Android

### What version of the app are you using?

1.95.0.441

**Comment by gaearon** [comment]
Yeah this is intentional. The heuristic is well show a reply if you follow the replier *and* at least one other person in the visible context.

I’m not opposed to adding an option to restrict it.

**Comment by nielsle** [comment]
Ideally the  "followed"  feed  should  only  show those threads  that have been  started by  a "followed  user".
And the  replies should  be collapsed until clicked.

When bluesky shows replies without context, then  the feed is often hard to  read (This happens when I  search  for a hashtag)
